File: Count and Say.cpp
class Solution {
public:
    string getNextCount(string str) {
        if (str == "") return "1";
        
        string result;
        for (int i = 0; i < str.length(); i++) {
            char ch = str[i];
            int cnt = 1;
            while (i < str.length() && ch == str[i+1]) {
                cnt++;
                i++;
            }
            result += (cnt + '0');
            result += ch;
        }
        return result;
    }
    string countAndSay(int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        string s = "";
        for (int i = 0; i < n; i++) {
            s = getNextCount(s);
        }
        
        return s;
    }
};

File: Count and Say.cs
public class Solution {
    public string CountAndSay(int n) {
        var str = "";
        for (var i = 0; i < n; i++) {
            str = GetNextCount(str);
        }
        
        return str;
    }
    
    private string GetNextCount(string str) {
        if (str == "") {
            return "1";
        }
        
        var sb = new StringBuilder();
        for (var i = 0; i < str.Length; i++) {
            var ch = str[i];
            var cnt = 1;
            while (i < str.Length - 1 && ch == str[i + 1]) {
                cnt++;
                i++;
            }
            sb.Append((char) (cnt + '0')).Append(ch);
        }
        return sb.ToString();
    }
}
